dotcom-rendering/src/web/components/LiveBlogEpic.importable.tsx

```diff
@@ -0,0 +1,289 @@
+import React, { useEffect, useState } from 'react';
+import { css } from '@emotion/react';
+import { space } from '@guardian/source-foundations';
+import { onConsentChange } from '@guardian/consent-management-platform';
+import { ConsentState } from '@guardian/consent-management-platform/dist/types';
+import { CountryCode, getCookie, joinUrl, log, storage } from '@guardian/libs';
+import { ModuleDataResponse } from '@guardian/support-dotcom-components';
+import {
+	getLastOneOffContributionTimestamp,
+	isRecurringContributor,
+	shouldHideSupportMessaging,
+} from '../lib/contributions';
+import { getLocaleCode } from '../lib/getCountryCode';
+import { setAutomat } from '../lib/setAutomat';
+import { useApi } from '../lib/useApi';
+
+type Props = {
+	section: string;
+	shouldHideReaderRevenue: boolean;
+	isPaidContent: boolean;
+	tags: TagType[];
+	contributionsServiceUrl: string;
+};
+
+const useConsent = () => {
+	const [consent, setConsent] = useState<ConsentState>();
+	const hasOptedOut = getCookie({ name: 'gu_article_count_opt_out' });
+
+	useEffect(() => {
+		if (hasOptedOut) return;
+		onConsentChange((state) => {
+			setConsent(state);
+		});
+	}, [hasOptedOut]);
+
+	return consent;
+};
+
+const useCountryCode = () => {
+	const [countryCode, setCountryCode] = useState<CountryCode | null>();
+
+	useEffect(() => {
+		getLocaleCode()
+			.then((cc) => {
+				setCountryCode(cc);
+			})
+			.catch((e) => {
+				const msg = `Error fetching country code: ${e}`;
+				window.guardian.modules.sentry.reportError(
+					new Error(msg),
+					'liveblog-epic',
+				);
+			});
+	}, []);
+
+	return countryCode;
+};
+
+const useEpic = ({ url, name }: { url: string; name: string }) => {
+	// Using state here to store the Epic component that gets imported allows
+	// us to render it with React (instead of inserting it into the dom manually)
+	const [Epic, setEpic] = useState<React.FC<{ [key: string]: unknown }>>();
+
+	useEffect(() => {
+		setAutomat();
+		window
+			.guardianPolyfilledImport(url)
+			.then((epicModule) => {
+				// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+				setEpic(() => epicModule[name]); // useState requires functions to be wrapped
+			})
+			.catch((err) => {
+				const msg = `Error importing LiveBlog epic: ${err}`;
+				window.guardian.modules.sentry.reportError(
+					new Error(msg),
+					'liveblog-epic',
+				);
+			});
+	}, [url, name]);
+
+	return { Epic };
+};
+
+function consentGivenForArticleCounts(consent: ConsentState): boolean {
+	const REQUIRED_CONSENTS_FOR_ARTICLE_COUNT = [1, 3, 7];
```

Comment:
What do these numbers mean? Is it worth defining these as constants somewhere?

Reply:
this is actually defined [here](https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/src/web/lib/contributions.ts#L146) as well.
Though I don't know specifically what these numbers mean, there must be a CMP guide somewhere...

Reply:
Also, I think we can use [this function](https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/src/web/lib/contributions.ts#L157) instead here

Reply:
I don't know what they mean (I think it's a SourcePoint thing?) but I've now made a change to import them from `lib/contributions`

Reply:
> Also, I think we can use [this function](https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/src/web/lib/contributions.ts#L157) instead here

That function returns a promise, which doesn't play nice with React. Also, the pattern here has `consent` being passed in as a prop rather than the function getting it itself. It was for these reasons I recreated it locally.

Reply:
I remember @mxdvl sent me a link one time to some json where you could see what all of the consents meant. It would be nice to have meaningful labels for these, but I guess that should live in the cmp repo instead of in dcr?